### PR TITLE
Add capability for roles api in files_sharing

### PR DESCRIPTION
**Pull request by sharidas**
Add capability for roles api in files_sharing

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

Add capabilty for roles api added in the server. `roles_api` is added to the public array to true, if `shareapi_enabled` is `yes` and `shareapi_allow_links` is `yes` set in the appconfig table.
Updated the change to arrange alignment of roles delivered from the RolesController and adjusted the description of View/Download/Edit in the roles. Updated the alignmnent and description change in the fontend (core/js/sharedialoglinkshareview.js).

## Related Issue

- Fixes <issue_link>

## Motivation and Context

Roles api can now be obtained from the capabilities. The key `roles_api` is added to true, if:
- `shareapi_enabled` is set to `yes`, in appconfig
- `shareapi_allow_links` is set to `yes`, in appconfig.

## How Has This Been Tested?

- php unit test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: apps/files_sharing/lib/Capabilities.php [hunk 1]
```diff
@@ -85,6 +85,8 @@ public function getCapabilities() {
 				$public['password']['enforced_for']['upload_only'] = $woPasswordEnforced;
 				$public['password']['enforced'] = $roPasswordEnforced || $rwPasswordEnforced || $woPasswordEnforced;
 
+				$public['roles_api_with_share_attributes'] = true;
```
Comment (PVince81) [inline apps/files_sharing/lib/Capabilities.php]: please rename to "roles_api"

**Comment by PVince81** [comment]
@sharidas as discussed we should also swap the permissions in this PR ?

File: apps/files_sharing/lib/Capabilities.php (hunk 1 above)
Comment (sharidas) [reply apps/files_sharing/lib/Capabilities.php]: Done.

**Comment by sharidas** [comment]
> @sharidas as discussed we should also swap the permissions in this PR ?

I have added a second commit to this PR. I was preparing the same and missed to add to this PR. Now I have added as  second commit.

**Comment by PVince81** [comment]
also needs swapping in the web UI accordingly ?

**Comment by sharidas** [comment]
> also needs swapping in the web UI accordingly ?

The swapping in the web UI is done.

File: apps/files_sharing/tests/CapabilitiesTest.php [hunk 2]
```diff
@@ -120,6 +120,29 @@ public function testNoLinkSharing() {
 		$this->assertFalse($result['public']['enabled']);
 	}
 
+	public function providesRolesCapbility() {
```
Comment (phil-davis) [inline apps/files_sharing/tests/CapabilitiesTest.php]:
```suggestion
	public function providesRolesCapability() {
```

File: apps/files_sharing/tests/CapabilitiesTest.php [hunk 3]
```diff
@@ -120,6 +120,29 @@ public function testNoLinkSharing() {
...
+		return [
+			[['core', 'shareapi_enabled', 'yes', 'no'],['core', 'shareapi_allow_links', 'yes', 'no']],
+			[['core', 'shareapi_enabled', 'yes', 'yes'],['core', 'shareapi_allow_links', 'yes', 'no']],
+			[['core', 'shareapi_enabled', 'yes', 'yes'],['core', 'shareapi_allow_links', 'yes', 'yes']]
+		];
+	}
+
+	/**
+	 * @dataProvider providesRolesCapbility
```
Comment (phil-davis) [inline apps/files_sharing/tests/CapabilitiesTest.php]:
```suggestion
	 * @dataProvider providesRolesCapability
```

File: apps/files_sharing/tests/CapabilitiesTest.php (hunk 2 above)
Comment (sharidas) [reply apps/files_sharing/tests/CapabilitiesTest.php]: Done

File: apps/files_sharing/tests/CapabilitiesTest.php (hunk 3 above)
Comment (sharidas) [reply apps/files_sharing/tests/CapabilitiesTest.php]: Done

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**